spec/granite/form/model/validations/nested_spec.rb

```diff
@@ -143,6 +143,106 @@
       it do
         expect { subject.valid? }.not_to raise_error
         expect(subject).not_to be_valid
+        expect(subject.errors.count).to eq(1)
+      end
+    end
+  end
+
+  context 'object field is invalid and has representation of that field' do
+    before do
+      stub_class(:validated_object) do
+        include Granite::Form::Model::Validations
+        include Granite::Form::Model::Attributes
+        include Granite::Form::Model::Primary
+
+        primary :id
+        attribute :title, String
+        validates_presence_of :title
+      end
+
+      Main.class_eval do
+        include Granite::Form::Model::Associations
+        include Granite::Form::Model::Representation
+        represents :title, of: :object
```

Comment:
You don't really use this `represents`, can we simulate the same situation by using `validates :object, nested: true` twice?

Reply:
No, the whole thing is that `represents` along with `validates :object, nested: true` validates it twice and doesn't clear errors as it should do between validations. I guess this logic is placed somewhere here: https://github.com/toptal/granite-form/blob/4a40c89e95eb5d26d83937ec164b9b6e8ecd7910/lib/granite/form/model/attributes/reflections/represents.rb#L14, but I haven't digged into that deeply.

Reply:
Yeah as you can see it just calls `target.validates_nested(options[:of])`, so I think we should be able to:
```
  validates :object, nested: true
  validates :object, nested: true
```

Which would simulate the same situation, right?